ludwig/hyperopt/execution.py

```diff
@@ -358,27 +358,21 @@ def _remove_partial_checkpoints(trial_path: str):
                 os.remove(marker_path)
 
     @contextlib.contextmanager
-    def _get_best_model_path(self, trial_path: str, analysis: ExperimentAnalysis) -> str:
-        remote_checkpoint_dir = self._get_remote_checkpoint_dir(Path(trial_path))
-        if remote_checkpoint_dir is not None:
-            self.sync_client.sync_down(remote_checkpoint_dir, trial_path)
-            self.sync_client.wait_or_retry()
-        self._remove_partial_checkpoints(trial_path)  # needed by get_best_checkpoint
-
-        try:
-            checkpoint = analysis.get_best_checkpoint(trial_path.rstrip("/"))
-        except Exception:
-            logger.warning(
-                f"Cannot get best model path for {trial_path} due to exception below:\n{traceback.format_exc()}"
-            )
-            yield None
-            return
+    def _get_best_model_path(self, trial_path: str, analysis: ExperimentAnalysis, creds: Dict[str, Any]) -> str:
+        checkpoint = analysis.get_best_checkpoint(trial=trial_path)
+        if checkpoint is None:
+            logger.warning("No best model found")
+            return None
 
-        if checkpoint is not None:
-            with checkpoint.as_directory() as path:
-                yield path
+        ckpt_type, ckpt_path = checkpoint.get_internal_representation()
+        if ckpt_type == "uri":
+            # Read remote URIs using Ludwig's internal remote file loading APIs, as
+            # Ray's do not handle custom credentials at the moment.
+            with use_credentials(creds):
+                yield ckpt_path
```

Comment:
`ckpt_path` here is a remote URL, so we'll want to convert to a directory before sending it back to the caller:

```
with use_credentials(creds):
    with checkpoint.as_directory() as path:
        yield path
```

Reply:
Fixed